### PR TITLE
APEXMALHAR-2412 Provide emitTuple overriding functionality for user in kinesis Input operator

### PR DESCRIPTION
**Pull request by deepak-narkhede**
Provide emitTuple overriding functionality for user in kinesis Input operator.

Tested with custom app. Will open another request for custom application as it is part of examples repo.

### CODE REVIEW

**Comment by deepak-narkhede** [comment]
@chaithu14 . Please review.

File: contrib/src/main/java/com/datatorrent/contrib/kinesis/AbstractKinesisInputOperator.java [hunk 1]
```diff
@@ -150,14 +150,11 @@ public AbstractKinesisInputOperator()
...
   /**
-   * Derived class has to implement this method, so that it knows what type of message it is going to send to Malhar.
-   * It converts a ByteBuffer message into a Tuple. A Tuple can be of any type (derived from Java Object) that
-   * operator user intends to.
-   *
-   * @param rc Record to convert into tuple
+   * Any concrete class derived from AbstractKinesisInputOperator to implement this method to emit tuples to an output port.
    */
-  public abstract T getTuple(Record rc);
+  protected abstract void emitTuple(Record rc);
```
Comment (chaithu14) [inline contrib/src/main/java/com/datatorrent/contrib/kinesis/AbstractKinesisInputOperator.java]: Why getTuple(..) method deleted ? This comes under backward compatible.
I would suggest define the emitTuple method as below:
```java
protected T emitTuple(Pair<String, Record> data)
{
  return getTuple(data.getSecond());
}
```

Comment (deepak-narkhede) [reply contrib/src/main/java/com/datatorrent/contrib/kinesis/AbstractKinesisInputOperator.java]: @chaithu14 .As discussed to balance out backward compatibility and new overriding functionality will keep existing getTuple() too.  Done with the changes.

**Comment by chaithu14** [comment]
Looks good to me.

**Pull request merged**